### PR TITLE
Add Author/License to LinuxComponent

### PR DESCRIPTION
**Pull request by sebasgomez238**
Adds Author and License properties from Syft output into the LinuxComponent.

### CODE REVIEW

File: src/Microsoft.ComponentDetection.Contracts/TypedComponent/LinuxComponent.cs [hunk 1]
```diff
@@ -9,12 +9,14 @@ private LinuxComponent()
         /* Reserved for deserialization */
     }
 
-    public LinuxComponent(string distribution, string release, string name, string version)
+    public LinuxComponent(string distribution, string release, string name, string version, string license = null, string author = null)
```
Comment (JamieMagee) [inline src/Microsoft.ComponentDetection.Contracts/TypedComponent/LinuxComponent.cs:12]: Can these be nullable? `string?`. Same for the properties below.

File: src/Microsoft.ComponentDetection.Detectors/linux/Contracts/SyftOutput.cs [hunk 2]
```diff
@@ -371,6 +371,22 @@ public partial struct License
 
     public static implicit operator License(string String) => new License { String = String };
     public static implicit operator License(string[] StringArray) => new License { StringArray = StringArray };
+
+    // Override ToString method to provide a string representation of the struct
+    public override string ToString()
```
Comment (JamieMagee) [inline src/Microsoft.ComponentDetection.Detectors/linux/Contracts/SyftOutput.cs:376]: This whole file is autogenerated. I would add a `SyftExtensions` file or class, and add an extension method for `License` instead.

Comment (sebasgomez238) [reply src/Microsoft.ComponentDetection.Detectors/linux/Contracts/SyftOutput.cs]: Turns out this was actually unnecessary. I Removed the changes in this file.

File: src/Microsoft.ComponentDetection.Contracts/TypedComponent/LinuxComponent.cs (hunk 1 above)
Comment (sebasgomez238) [reply src/Microsoft.ComponentDetection.Contracts/TypedComponent/LinuxComponent.cs]: They can be but the analyzer kept complaining. This was the only way I could get it to not show errors. I didn't want to use "#nullable enable" for the entire file.

**Review by JamieMagee — APPROVED** [review]

**Review by cobya — APPROVED** [review]

**Pull request merged**